pkg/asset/tls/certkey.go

```diff
@@ -0,0 +1,153 @@
+package tls
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"net"
+	"path/filepath"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// CertKey contains the private key and the cert that's
+// signed by the parent CA.
+type CertKey struct {
+	// assetStock is a reference to the store so that
+	// we can get the InstallConfig instance.
+	assetStock *asset.Stock
+
+	// Common fields.
+	Subject      pkix.Name
+	Validity     time.Duration
```

Comment:
If everyone is using 10-years for this, we can drop this knob to simplify the structure.

Reply:
>  If everyone is using 10-years for this...

[They aren't][1], so this property is fine as it stands.

[1]: https://github.com/openshift/installer/pull/145#discussion_r211052124